File: PF2-web5/web5-vue/web2-vue/src/types/Intent.ts
export class Intent {
  intent_no: number;
  intent_context: string;
  intent_shortname: string;
  intent_x: number;
  intent_y: number;
  intent_h: number;
  intent_w: number;

  static newIntent(no: number, name: string, shortname: string, x: number, y: number, h: number, w: number) {
    let req = new Intent();
    req.intent_no = no;
    req.intent_context = name;
    req.intent_shortname = shortname;
    req.intent_x = x;
    req.intent_y = y;
    req.intent_h = h;
    req.intent_w = w;
    return req;
  }

  static newIntentWithOld(old: Intent, name: string, shortname: string) {
    let req = new Intent();
    req.intent_no = old.intent_no;
    req.intent_context = name;
    req.intent_shortname = shortname;
    req.intent_x = old.intent_x;
    req.intent_y = old.intent_y;
    req.intent_h = old.intent_h;
    req.intent_w = old.intent_w;
    return req;
  }

  static copyIntent(old: Intent) {
    let req = new Intent();
    req.intent_no = old.intent_no;
    req.intent_context = old.intent_context;
    req.intent_shortname = old.intent_shortname ? old.intent_shortname : getShortname(old.intent_context);
    req.intent_x = old.intent_x;
    req.intent_y = old.intent_y;
    req.intent_h = old.intent_h;
    req.intent_w = old.intent_w;
    return req;
  }

  getNo() { return this.intent_no; }
  setNo(no: number) { this.intent_no = no; }
  getName() { return this.intent_context; }
  setName(name: string) { this.intent_context = name; }
  getShortName() { return this.intent_shortname; }
  setShortName(shortName: string) { this.intent_shortname = shortName; }
  getX() { return this.intent_x; }
  setX(x: number) { this.intent_x = x; }
  getY() { return this.intent_y; }
  setY(y: number) { this.intent_y = y; }
  getH() { return this.intent_h; }
  setH(h: number) { this.intent_h = h; }
  getW() { return this.intent_w; }
  setW(w: number) { this.intent_w = w; }
}

function getShortname(name: string) {
  if (!name) return "";
  return name.replace(/\s+/g, "");
}

File: PF2-web5/web5-backend/src/main/java/com/example/demo/bean/Intent.java
package com.example.demo.bean;

public class Intent extends PdNode{
	private int intent_no;	//需求编号
	private String intent_context;	//需求描述
	private String intent_shortname;
	private int intent_x;	//位置信息
	private int intent_y;
	private int intent_h;
	private int intent_w;
	public Intent(Intent item) {
		intent_no = item.intent_no;
		intent_context = item.intent_context;
		intent_shortname = item.intent_shortname;
		intent_x = item.intent_x;
		intent_y = item.intent_y;
		intent_h = item.intent_h;
		intent_w = item.intent_w;
	}

	public Intent() {
		super();
	}

	public int getIntent_no() {
		return intent_no;
	}
	public void setIntent_no(int intent_no) {
		this.intent_no = intent_no;
	}
	public String getIntent_context() {
		return intent_context;
	}
	public void setIntent_context(String intent_context) {
		this.intent_context = intent_context;
	}
	public String getIntent_shortname() {
		return intent_shortname;
	}
	public void setIntent_shortname(String intent_shortname) {
		this.intent_shortname = intent_shortname;
	}
	public int getIntent_x() {
		return intent_x;
	}
	public void setIntent_x(int intent_x) {
		this.intent_x = intent_x;
	}
	public int getIntent_y() {
		return intent_y;
	}
	public void setIntent_y(int intent_y) {
		this.intent_y = intent_y;
	}
	public int getIntent_h() {
		return intent_h;
	}
	public void setIntent_h(int intent_h) {
		this.intent_h = intent_h;
	}
	public int getIntent_w() {
		return intent_w;
	}
	public void setIntent_w(int intent_w) {
		this.intent_w = intent_w;
	}
	public int getNo() {
		return intent_no;
	}

	public void setNo(int no) {
		this.intent_no = no;
	}

	public String getName() {
		return intent_context;
	}

	public void setName(String name) {
		this.intent_context = name;
	}

	public String getShortname() {
		return intent_shortname;
	}

	public void setShortname(String shortname) {
		this.intent_shortname = shortname;
	}

	public int getX() {
		return intent_x;
	}

	public void setX(int x) {
		this.intent_x = x;
	}

	public int getY() {
		return intent_y;
	}

	public void setY(int y) {
		this.intent_y = y;
	}

	public int getH() {
		return intent_h;
	}

	public void setH(int h) {
		this.intent_h = h;
	}

	public int getW() {
		return intent_w;
	}

	public void setW(int w) {
		this.intent_w = w;
	}
}
